Packs/CheckPointHarmonyEndpoint/Integrations/CheckPointHarmonyEndpoint/CheckPointHarmonyEndpoint.py

```diff
@@ -0,0 +1,2703 @@
+        args["job_id"] = response.get("jobId")
+
+    return schedule_command(args, client, demisto.command())
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get("interval", 30)),
+    timeout=arg_to_number(demisto.args().get("timeout", 600)),
+    poll_message="Computer shutdown request is executing",
+    requires_polling_arg=False,
+)
+def computer_shutdown_command(args: Dict[str, Any], client: Client) -> PollResult:
+    """Shuts-down computers match the given query.
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        request_body["operationParameters"] |= {
+            "forceAppsShutdown": args.get("force_apps_shutdown"),
+        }
+        response = client.computer_shutdown(request_body)
+        args["job_id"] = response.get("jobId")
+
+    return schedule_command(args, client, demisto.command())
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get("interval", 30)),
+    timeout=arg_to_number(demisto.args().get("timeout", 600)),
+    poll_message="Computer repair request is executing",
+    requires_polling_arg=False,
+)
+def computer_repair_command(args: Dict[str, Any], client: Client) -> PollResult:
+    """Repairs the Harmony Endpoint Client installation on computers matching the given query.
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        response = client.computer_repair(request_body)
+        args["job_id"] = response.get("jobId")
+
+    return schedule_command(args, client, demisto.command())
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get("interval", 30)),
+    timeout=arg_to_number(demisto.args().get("timeout", 600)),
+    poll_message="Computer list fetch request is executing",
+    requires_polling_arg=False,
+)
+def computer_list_command(args: Dict[str, Any], client: Client) -> PollResult:
+    """Gets a list of computers matching the given filters.
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        updated_request_body = {
+            "filters": request_body["targets"]["query"]["filter"],
+            "paging": request_body["targets"]["query"]["paging"],
+        }
+        del request_body
```

Comment:
Instead of doing this, add a flag argument to the `build_request_body` with default value of `False` that will indicate if we want to return just the `request_body["targets"]["query"]` value instead of the whole dict. In the `build_request_body` function, save this value to an argument and return it in case the flag value is `True`.

Reply:
I didn't get you here

Reply:
I understand your meaning.
Instead of using a flag for one scenario out of 21, I have extracted the relevant logic for this specific use case.